File: code/fixture.js
"use strict";

var argv = process.argv

var usage = function() {
  console.log("usage:"
              + argv[0]
              + argv[1]
              + "<solution> [case]");
  process.exit(-1);
}

if (argv.length < 3 || argv.length > 4) {
  usage();
};

var solution_file = "./implementations/" + argv[2] + ".js";

console.log('requiring ' + solution_file);
var solution = require(solution_file).solution;

var cases = require('./cases.json');

var try_case = function(i) {

  var thiscase = cases[i];
  var pirates = thiscase.pirates;
  var answer = thiscase.answer;

  console.log();
  console.log("trying testcase " + i + ":", pirates);

  var myanswer = solution(pirates);

  console.log("my answer: " + myanswer + ". correct answer: " + answer);

};

if (argv.length == 4) {
  try_case(argv[3]);
} else {
  for (var i=0; i<cases.length; i++) {
    try_case(i);
}};

File: code/fixture.py
from sys import argv, exit
import json
import imp


def usage():
    print 'usage: {} <solution> [case]'.format(argv[0])
    exit(-1)

if len(argv) < 2 or len(argv) > 3:
    usage()

fn = argv[1]
n = int(argv[2]) if len(argv) == 3 else None

solution = imp.load_module(
    fn, *imp.find_module(
        fn, ['implementations']
    )).solution

with open('cases.json') as casesfile:
    cases = json.load(casesfile)


def try_case(i):

    case = cases[i]
    pirates = case['pirates']
    answer = case['answer']

    print
    print 'trying testcase {}: {}'.format(i, pirates)

    myanswer = solution(pirates)

    print 'my answer: {}. correct answer: {}'.format(
        myanswer,
        answer
    )

for i in [n] if n is not None else range(len(cases)):
    try_case(i)
